src/api/app/controllers/public_controller.rb

```diff
@@ -200,7 +200,9 @@ def check_package_access(project_name, package_name, use_source = true)
     end
 
     # generic access checks
-    key = 'public_package:' + project_name + ':' + package_name
+    project = Project.find_by(name: project_name)
```

Comment:
access checks like `.get_by_name` or similar?

Reply:
The thing is that this before action is caching the result of an access check (when sources are not checked). I was hoping to avoid calling any expensive methods to get a cache key.
Which is a bit tricky given that there are remote projects and project links.

But I think `project.try(:cache_key) || project_name` all this to a satisfiable degree:thinking:

Reply:
Talked about this offline. Using `.find_by` is okay in this case because what happens inside the `Rails.cache.fetch` is doing access check again...